src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -468,8 +469,7 @@ def shape(self) -> tuple:
         """
         return (len(self),)
 
-    @is_lazy_check
-    def astype(self, dtype: Any, copy: bool = True) -> Index:
+    def astype(self, dtype: str | type | ExtensionDtype, copy: bool = True) -> Index:
```

Comment:
did we double check that the generated doc have the correct Index class link?

Reply:
Yup! Works as expected